src/SprykerSdk/Zed/ComposerConstrainer/Business/Finder/UsedForeignModuleFinder/UsedForeignModuleFinder.php

```diff
@@ -0,0 +1,199 @@
+    {
+        $fileContent = $splFileInfo->getContents();
+        foreach ($this->getUsedClassesInFile($fileContent) as $usedClassName) {
+            if (!$this->isExcludedClass($usedClassName) && $this->isVendorClass($usedClassName)) {
+                $usedModuleTransfer = $this->getUsedModuleByClassName($usedClassName);
+                $usedModuleCollectionTransfer->addUsedModule($usedModuleTransfer);
+            }
+        }
+
+        return $usedModuleCollectionTransfer;
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    protected function isExcludedClass(string $className): bool
+    {
+        $namespaces = array_merge(
+            $this->config->getCoreNamespaces(),
+            $this->config->getExcludedNameSpaces(),
+            $this->config->getProjectNamespaces()
+        );
+        $pattern = sprintf('/(%s)\\\\/', implode('|', $namespaces));
+
+        return (bool)preg_match_all($pattern, $className);
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\AbstractTransfer\UsedModuleTransfer|null
+     */
+    protected function getUsedModuleByClassName(string $className): ?UsedModuleTransfer
+    {
+        $classFilename = $this->getClassFileNameByClassName($className);
+        $composerJsonData = $this->getComposerJsonDataByClassFilename($classFilename);
+        $packageName = explode('/', $composerJsonData['name']);
+        $usedModuleTransfer = (new UsedModuleTransfer())
+            ->setOrganization($packageName[0])
+            ->setModule($packageName[1]);
+
+        return $usedModuleTransfer;
+    }
+
+    /**
+     * @param string $classFilename
+     *
+     * @return array
+     */
+    protected function getComposerJsonDataByClassFilename(string $classFilename): array
+    {
+        $pattern = sprintf('/(%s)([^\/]+\/+){2}/', str_replace('/', '\/', $this->config->getVendorDirectory()));
+        preg_match($pattern, $classFilename, $matches);
+        $composerJsonFilePath = $matches[0];
+        $composerJsonData = $this->composerJsonReader->readFromFilePath($composerJsonFilePath);
+
+        return $composerJsonData;
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    protected function isVendorClass(string $className): bool
+    {
+        $filename = $this->getClassFileNameByClassName($className);
+        $pattern = sprintf('/%s/', str_replace('/', '\/', $this->config->getVendorDirectory()));
+
+        return (bool)preg_match($pattern, $filename);
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return string
+     */
+    protected function getClassFileNameByClassName(string $className): string
+    {
+        return (new ReflectionClass($className))->getFileName();
+    }
+
+    /**
+     * @param string $fileContent
+     *
+     * @return string[]
+     */
+    protected function getUsedClassesInFile(string $fileContent): array
```

Comment:
I would say ClassNames describes the result of this function a little bit better

Reply:
Updated.